pkgs/applications/audio/open-stage-control/default.nix

```diff
@@ -33,14 +30,19 @@ nodeComposition.package.override rec {
 
   doInstallCheck = true;
 
-  preRebuild = ''
-    # remove electron to prevent building since nixpkgs electron is used instead
-    rm -r node_modules/electron
-  '';
+  makeCacheWritable = true;
+  npmFlags = [ "--legacy-peer-deps" ];
+
+  # Override installPhase so we can copy the only folders that matter (app and node_modules)
+  installPhase = ''
+    runHook preInstall
 
-  postInstall = ''
-    # build assets
-    npm run build
+    # prune unused deps
+    npm prune --omit dev $npmFlags
```

Comment:
Can't we do this when generating the lock file?

Reply:
No because then the dev dependencies wouldn't be available for the build step

Reply:
The `npmInstallHook` actually does the prune like that too (it's standard-ish for npm builds), but we needed a custom install phase here anyway for the reasons in the comment above the phase and because the hook doesn't currently respect `npmFlags` for the prune step (although that has been fixed in an open follow-up PR)